crates/shared/src/sources/uniswap_v3/event_fetching.rs

```diff
@@ -1,99 +1,63 @@
     std::collections::BTreeMap,
 };
 
-const SWAP_TOPIC: [u8; 32] =
-    hex!("c42079f94a6350d7e6235f29174924f928cc2ac818eb64fed8004e115fbcca67");
-const BURN_TOPIC: [u8; 32] =
-    hex!("0c396cd989a39f4459b5fa1aed6a9a8dcdbc45908acfd67e028cd568da98982c");
-const MINT_TOPIC: [u8; 32] =
-    hex!("7a53080ba414158be7ec69b987b5fb7d07dee101fe85488f0853ae16239d0bde");
-
-#[derive(Clone, Debug, Eq, PartialEq)]
-pub enum UniswapV3Event {
-    Burn(Burn),
-    Mint(Mint),
-    Swap(Swap),
-}
+pub struct UniswapV3PoolEventFetcher(pub DynProvider);
 
-impl ParseLog for UniswapV3Event {
-    fn parse_log(log: RawLog) -> Result<Self, ExecutionError> {
-        let standard_event: Option<Result<UniswapV3Event, ExecutionError>> =
-            log.topics.first().copied().map(|topic| match topic {
-                H256(BURN_TOPIC) => Ok(UniswapV3Event::Burn(
-                    log.clone().decode(
-                        UniswapV3Pool::raw_contract()
-                            .interface
-                            .abi
-                            .event("Burn")
-                            .expect("generated event decode"),
-                    )?,
-                )),
-                H256(MINT_TOPIC) => Ok(UniswapV3Event::Mint(
-                    log.clone().decode(
-                        UniswapV3Pool::raw_contract()
-                            .interface
-                            .abi
-                            .event("Mint")
-                            .expect("generated event decode"),
-                    )?,
-                )),
-                H256(SWAP_TOPIC) => Ok(UniswapV3Event::Swap(
-                    log.clone().decode(
-                        UniswapV3Pool::raw_contract()
-                            .interface
-                            .abi
-                            .event("Swap")
-                            .expect("generated event decode"),
-                    )?,
-                )),
-                _ => Err(ExecutionError::from(Error::InvalidData)),
-            });
-        if let Some(Ok(data)) = standard_event {
-            return Ok(data);
-        }
-        Err(ExecutionError::from(Error::InvalidData))
+impl AlloyEventRetrieving for UniswapV3PoolEventFetcher {
+    type Event = AlloyUniswapV3PoolEvents;
+
+    fn filter(&self) -> Filter {
+        // No pool address filter since the generated request might be too large
+        // leading to RPC performance issues.
+        // More details: <https://github.com/cowprotocol/services/pull/3620>
+        let mut set = FilterSet::default();
+        set.insert(Swap::SIGNATURE_HASH);
+        set.insert(Burn::SIGNATURE_HASH);
+        set.insert(Mint::SIGNATURE_HASH);
+        Filter::new().address(vec![]).event_signature(set)
+    }
+
+    fn provider(&self) -> &DynProvider {
+        &self.0
     }
 }
 
-pub struct UniswapV3PoolEventFetcher(pub Web3);
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct WithAddress<T>(T, Address);
```

Comment:
Is this just a container to carry an address around with T? A brief doc would eliminate doubts